### ISSUE TITLE
[enhancement] Make autosubmit run detach by default, avoid needing nohup/nohup.log

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1243]
Hello @dbeltrankyl and @mcastril ,

At the moment users launch Autosubmit with `autosubmit run`, which blocks the terminal. The recommended way to launch it to the background is with `nohup`. It's in our documentation, and that's how most users launch it.

Some issues with this approach are

a) only the user launching the command has access to the nohup.log (when a robot account is used, AS developers not using the account are "locked out" and cannot see that file, regardless of umask/acl/linux permissions)
b) when the user forgets to `nohup`, they need to kill the command and launch it again (even if it might be doable to send it to the background, it is probably safer to avoid it)

I suggest we copy the behavior of Cylc here, and `autosubmit run` automatically sends the command to the background, and the logs to a file in the AS log files. That should simplify operation a bit, and allow users to run it without having to use `nohup`.

Furthermore, we can also copy the `--no-detach` flag so that `autosubmit run --no-detach $expid` keeps the terminal blocked, if the user so desires.

**Comment by kinow** [comment]
mentioned in issue digital-twins/de_340/project_management#632

**Comment by eferre1** [comment]
Hi @kinow, I guess this is still in the backlog, but just a suggestion for it, to also have an option to redirect the log output (if possible). 

This is my use case, I usually run multiple experiments in parallel (either testing suite, or working on different tasks while waiting results from another exp), and then redirect all the logs to the same file, so I only have one terminal with a `tail -f logs_file.txt`. Some other users that I know, redirect the log adding a number to know the attempt (in case something fails, they can check the `nohupX.out` for further references).

**Comment by kinow** [comment]
Hi @@eferre1 

> I guess this is still in the backlog, 

Correct. This issue hasn't been assigned any milestone yet, and when I first proposed it, it created some discussion on whether to break the current behaviour or not.


> but just a suggestion for it, to also have an option to redirect the log output (if possible).
> 
> This is my use case, I usually run multiple experiments in parallel (either testing suite, or working on different tasks while waiting results from another exp), and then redirect all the logs to the same file, so I only have one terminal with a tail -f logs_file.txt. Some other users that I know, redirect the log adding a number to know the attempt (in case something fails, they can check the nohupX.out for further references).

Thanks for the suggestion! My idea for this is to mimic how ecflow & cylc work, where users & operators fire a suite/workflow without having to block their terminals, nor having to use `nohup`. And indeed, all the logs will be kept automatically on disk, so that you can `tail -f` the files 👍 

Thanks!

p.s.: a big advantage of doing it this different way, is that it's a lot easier to automate it as you can easily issue commands like `autosubmit run` without having to fiddle with nohups, redirect streams, etc...

**Comment by sparonuz** [comment]
Hi @kinow, sorry to jump in!

With Quentin (Gaudel), in the framework of  EDITO, we are looking for an autosubmit feature which looks very similar to what you are describing here! But I would like to confirm, and open a new issue if needed (and sensible).

So: we created a process that runs a simple two step workflow: it pulls a docker image on the login, and runs it on a compute node. We would like to be able, directly from the edito interface, to capture the ouptut (so the singularity pull and singularity run output) so that we would not be forced to transfer the log files from the HPC to the EDITO S3 after autosubmit finishes. So basically having autosubmit streaming the logs output to the std output.

Do you think that the feature you're talking about in this issue would satisfy this need, or that is something different? Do you think it is even possible what I am asking for?

Best regards, 
Stella

**Comment by kinow** [comment]
Hi @sparonuz , you are always welcome to join in any issues or PRs!

> Do you think that the feature you're talking about in this issue would satisfy this need, or that is something different? Do you think it is even possible what I am asking for?

Not this feature, no. This issue is about changing Autosubmit to what Cylc does by default when you `cylc play flow` in your terminal, and you get a background process running (in detached mode). Or if you want to block the terminal, do something similar to `cylc play --no-detach flow`, or `autosubmit run --no-detach a001`.

What you are describing, **maybe** could work with `autosubmit cat-log a000 ... workflow-task-that-runs-ansible`. I copied the cat-log command to Autosubmit some time ago, but it's still missing exactly remote tasks (#1049). With that, you could maybe use a `cat-log -mode f` at the end of the script, or `-mode t` (if memory serves me well, the first dumps once the file, and the mode tail keeps updating the stdout), and then get the output of the logs in your stream/terminal.

If you think that would work we can check the priority of that issue as it shouldn't be too hard and we now have three junior devs helping us who could start working on it in a short time.

**Comment by sparonuz** [comment]
> Not this feature, no. This issue is about changing Autosubmit to what Cylc does by default when you cylc play flow in your terminal, and you get a background process running (in detached mode). Or if you want to block the terminal, do something similar to cylc play --no-detach flow, or autosubmit run --no-detach a001.

You are right! I was probably confused by Eric's message. I will move this conversation to #1049, it looks exactly what I need. 

Stella
